messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -655,6 +669,32 @@ private void awaitTermination() {
         return CompletableFuture.runAsync(this::awaitTermination);
     }
 
+    private void processScopeEvents() throws Exception {
+        if (scopeEvents.size() == 0) {
+            return;
+        }
+        ScopeEvent scopeEvent;
+        Map<EventMessage<?>, ScopeDescriptor> batch = new LinkedHashMap<>();
+        while ((scopeEvent = scopeEvents.poll()) != null && batch.size() < batchSize) {
+            batch.put(scopeEvent.event, scopeEvent.scope);
+        }
+        BatchingUnitOfWork<EventMessage<?>> uow = new BatchingUnitOfWork<>(new ArrayList<>(batch.keySet()));
+        processInUnitOfWork(batch, uow);
```

Comment:
Gotcha, understanding why this is a `Map`! Please disregard the comment on the `SubscribingEventProcessor`.

Reply:
But I already explained 😢 😄